tests/shared/core/test_domain.py

```diff
@@ -2067,3 +2067,112 @@ def test_merge_domain_with_separate_session_config():
         domain.session_config.session_expiration_time
         == expected_session_expiration_time
     )
+
+
+@pytest.mark.parametrize(
+    "actions, expected_result",
+    [
+        (
+            [
+                {"action_hello_world": {"send_domain": False}},
+                {"action_say_something": {"send_domain": True}},
+                {"action_calculate": {"send_domain": True}},
+                "action_no_domain",
+            ],
+            ["action_say_something", "action_calculate"],
+        )
+    ],
+)
+def test_collect_actions_which_explicitly_need_domain(
+    actions: List[Union[Dict[Text, Any], str]], expected_result: List[str]
+):
+    result = Domain._collect_actions_which_explicitly_need_domain(actions)
+
+    # assert that two unordered lists have same elements
+    assert sorted(result) == sorted(expected_result)
+
+
+@pytest.mark.parametrize(
+    "actions, expected_result",
+    [
+        (
+            [
+                {"action_hello_world": {"send_domain": False}},
+                {"action_say_something": {"send_domain": True}},
+                {"action_calculate": {"send_domain": True}},
+                "action_no_domain",
+            ],
+            [
+                "action_hello_world",
+                "action_say_something",
+                "action_calculate",
+                "action_no_domain",
+            ],
+        )
+    ],
+)
+def test_collect_actions(
+    actions: List[Union[Dict[Text, Any], str]], expected_result: List[str]
+):
+    result = Domain._collect_actions(actions)
+
+    # assert that two unordered lists have same elements
+    assert sorted(result) == sorted(expected_result)
+
+
+@pytest.mark.parametrize(
+    "content, expected_user_actions, expected_actions_which_explicitly_need_domain",
+    [
+        (
+            """
+        intents:
```

Comment:
Please add `version` key to the domain contents to prevent overflow of UserWarnings in the CI.